docs/tutorials/gluon/info_gan.md

```diff
@@ -0,0 +1,438 @@
+loss1 = gluon.loss.SigmoidBinaryCrossEntropyLoss()
+loss2 = gluon.loss.L2Loss()
+loss3 = gluon.loss.SoftmaxCrossEntropyLoss()
+```
+
+This function samples `c`, `z`, and concatenates them to create the generator input.
+
+
+```python
+def create_generator_input():
+    
+    #create random noise
+    z      = nd.random_normal(0, 1, shape=(batch_size, z_dim), ctx=ctx)
+    label  = nd.array(np.random.randint(n_categories, size=batch_size)).as_in_context(ctx)
+    c1     = nd.one_hot(label, depth=n_categories).as_in_context(ctx)
+    c2     = nd.random.uniform(-1, 1, shape=(batch_size, n_continuous)).as_in_context(ctx)
+
+    # concatenate random noise with c which will be the input of the generator
+    return nd.concat(z, c1, c2, dim=1)
+```
+
+Define the training loop. 
+1. The discriminator receives `real_data` and `loss1` measures how many real images have been identified as real
+2. The discriminator receives `fake_image` from the Generator and `loss1` measures how many fake images have been identified as fake
+3. Update Discriminator. Currently, it is updated every second iteration in order to avoid that the Discriminator becomes too strong. You may want to change that.
+4. The updated discriminator receives `fake_image` and `loss1` measures how many fake images have been been identified as real, `loss2` measures the difference between the sampled continuous latent code `c` and the output of the Q model and `loss3` measures the difference between the sampled categorical latent code `c` and the output of the Q model.
+4. Update Generator and Q
+
+
+```python
+with SummaryWriter(logdir='./logs/') as sw:
+    
+    epochs = 1
+    i = 0
+    for epoch in range(epochs):
+        print("Epoch", epoch)
+        starttime = time.time()
+        
+        d_error_epoch = nd.zeros((1,), ctx=ctx)
+        g_error_epoch = nd.zeros((1,), ctx=ctx)
+        
+        for idx, data in enumerate(train_dataloader):
+            i = i + 1
+            
+            #get real data and generator input
+            real_data = data.as_in_context(ctx)     
+            g_input   = create_generator_input()
+
+            
+            #Update discriminator: Input real data and fake data
+            with autograd.record():
+                output_real,_,_ = discriminator(real_data)
+                d_error_real    = loss1(output_real, real_label)
+                
+                # create fake image and input it to discriminator
+                fake_image      = generator(g_input)
+                output_fake,_,_ = discriminator(fake_image.detach())
+                d_error_fake    = loss1(output_fake, fake_label)
+                
+                # total discriminator error
+                d_error         = d_error_real + d_error_fake
+
+            d_error_epoch += d_error.mean()
+            
+            #Update D every second iteration
+            if i % 2 == 0:
+                d_error.backward()
+                d_trainer.step(data.shape[0])
```

Comment:
you can define a descriptive variable name for data.shape[0] as it's used several times.